### PR TITLE
[FIX] spreadsheet: keeps link to Odoo menu when copying chart

### PR DESCRIPTION
**Pull request by anhe-odoo**
## Task Description:

When copying a chart in a spreadsheet, the link to the Odoo menu is not kept. This PR aims to fix it by storing the Odoo Menu in the new extraData key of the chart definition.

## Related Task/PR:

- Task-3380568
- https://github.com/odoo/o-spreadsheet/pull/2819

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js [hunk 1]
```diff
@@ -16,7 +16,27 @@ export default class ChartOdooMenuPlugin extends CorePlugin {
     handle(cmd) {
         switch (cmd.type) {
             case "LINK_ODOO_MENU_TO_CHART":
-                this.history.update("odooMenuReference", cmd.chartId, cmd.odooMenuId);
+                {
+                    this.history.update("odooMenuReference", cmd.chartId, cmd.odooMenuId);
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js:20]: We can totally drop `this.odooMenuReference` and use the information in the chart definition right ? So we won't have to handle `CREATE/DELETE_CHART`.

I'd also think that import/export the `odooMenuId` should be done by the chart when it export/import its definition, not by this plugin.

File: addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js [hunk 2]
```diff
@@ -16,10 +15,17 @@ export default class ChartOdooMenuPlugin extends CorePlugin {
     handle(cmd) {
         switch (cmd.type) {
             case "LINK_ODOO_MENU_TO_CHART":
-                this.history.update("odooMenuReference", cmd.chartId, cmd.odooMenuId);
-                break;
-            case "DELETE_FIGURE":
-                this.history.update("odooMenuReference", cmd.id, undefined);
+                {
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js:18]: really nitpick: the bracket is functionally useless, and this prettying of a line with a single bracket is ugly IMO.

File: addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js [hunk 3]
```diff
@@ -31,20 +37,18 @@ export default class ChartOdooMenuPlugin extends CorePlugin {
...
-        const menuId = this.odooMenuReference[chartId];
-        return menuId ? this.getters.getIrMenu(menuId) : undefined;
-    }
-
-    import(data) {
-        if (data.chartOdooMenusReferences) {
-            this.odooMenuReference = data.chartOdooMenusReferences;
+        try {
+            const definition = this.getters.getChartDefinition(chartId);
+            if (definition.extraData && definition.extraData.odooMenuId) {
```
Comment (rrahir) [inline addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js:42]: this probably implies a migration from previous versions, right?

Comment (anhe-odoo) [reply addons/spreadsheet/static/src/chart/plugins/chart_odoo_menu_plugin.js:42]: Yes ... :'(

**Pull request closed without merging**